File: app/src/main/java/com/example/requestapp/ui/fragment/editTaskFragment/EditTaskFragmentContract.java
package com.example.requestapp.ui.fragment.editTaskFragment;

import com.example.requestapp.model.Task;

public interface EditTaskFragmentContract {
    interface Presenter{

        void onClickedEdit(Task task,String descryption);

        void onClickedFinish(Task task);
    }
    interface View{

        void showMessage(String message);

        void lockEditTextDescryption();

        void unlockEditTextDescryption();

        void lockButtonSave();

        void unlockButtonSave();

        void changeNameButton(String name);

        void closeFragment();

    }
}

File: app/src/main/java/com/example/requestapp/ui/fragment/editTaskFrgment/EditTaskFragmentContract.kt
package com.example.requestapp.ui.fragment.editTaskFrgment

import com.example.requestapp.model.Task

interface EditTaskFragmentContract {
    interface Presenter {
        fun onClickedEdit(task: Task?, descryption: String?)
        fun onClickedFinish(task: Task?)
    }

    interface View {
        fun showMessage(message: String?)
        fun lockEditTextDescryption()
        fun unlockEditTextDescryption()
        fun lockButtonSave()
        fun unlockButtonSave()
        fun changeNameButton(name: String?)
        fun closeFragment()
    }
}